WebRoot/js/zimbraAdmin/accounts/model/ZaAccount.js

```diff
@@ -2533,6 +2533,11 @@ ZaAccount.isEmailRetentionPolicyEnabled = function () {
     try {
         var instance  = this.getInstance () ;
         var gc   = ZaApp.getInstance().getGlobalConfig();
+        if (!instance.attrs[ZaAccount.A_mailHost]) {
+            // GetEffectiveRightsRequest fails with an exception if a server name is blank
+            var ex = new ZmCsfeException("The request was not sent", ZmCsfeException.ACCT_INVALID_ATTR_VALUE, "GetEffectiveRightsRequest", "zimbraMailHost cannot be read");
+            throw(ex);
```

Comment:
in this case without checking getEffectiveRights, we are enabling email retention policy for delegated admin, is this desired behaviour?
shouldn't we need to disable email retention policy?

Reply:
Modified to hide them rather than disable them.
The reason is,
- the admin doesn't know whether the retention policy (=MailboxPurge) run on the server without getting server config
- if we disable them, the lifetime settings are shown as read only. Then the admin may misunderstand that the purge works even if server's zimbraMailPurgeSleepInterval is 0.

Reply:
Thanks